scripts/model/blog.js

```diff
@@ -42,27 +54,24 @@
         url: url,
         success: function(data, message, xhr){
           var newTag=xhr.getResponseHeader('eTag');
-          if (newTag !== localStorage['eTag'+name]){
-            localStorage[name] ='';
-            console.log('getting new  blog data');
+          if (newTag !== localStorage['eTag' + name]){
+            localStorage[name] = '';
+            console.log(newTag, localStorage['eTag' + name]);
+            console.log(name, 'getting new  blog data');
             Post.fetchAll(url, name, nextFunction); // recursive call
           } //end of if
-          console.log('got your blog right here');
-          var retreivedData =  JSON.parse(localStorage.getItem(name, data));
-          console.log(retreivedData);
-          Post.loadIntoObjectArray(retreivedData);
-          nextFunction();
+          else{
+            console.log('eTags the same', newTag, localStorage['eTag' + name]);
+            console.log('got your blog right here');
+            var retreivedData =    JSON.parse(localStorage.getItem(name));
+            console.log(retreivedData);
+            Post.loadIntoObjectArray(retreivedData);
+            nextFunction();
+          }
         } //end of success
       });  //end of ajax
     };
   };
 
```

Comment:
I pulled down your code, and I would need to spend more time with this `fetchAll()` function, but I am wondering if we could refactor/split some of the `fetchAll()` function into another function, like the `getAll()` function we had in class? Along with calling your `loadIntoObjectArray()` function and your `nextFunction()` function, the focus for a `getAll()` function would be on these lines: 

```
      $.get(url, function(data, message, xhr) {
        // console.log('data from get', data);
        localStorage.setItem(name,JSON.stringify(data));
        console.log(xhr.getResponseHeader('eTag'));
        localStorage['eTag'+name] = xhr.getResponseHeader('eTag');
        Post.fetchAll(url, name, nextFunction); // recursive call
```

and these lines

```
 console.log('got your blog right here');
          var retreivedData =  JSON.parse(localStorage.getItem(name, data));
          console.log(retreivedData);
          Post.loadIntoObjectArray(retreivedData);
          nextFunction();
```

Splitting the functions might prevent the need for a recursive call. Again, I'll need to play with your code some more to fully flesh this thought out with working code, but the current code works - so that is what is most important. Nice job! Overall, the `fetchAll()` function looks good and isn't too long. I wouldn't stress over refactoring at the moment.